### PR TITLE
#8902: add input and attn_mask del

### PR DESCRIPTION
**Pull request by sraizada-tt**
Need these to have good pcc with program cache enabled. Related issue: [7159](https://github.com/tenstorrent/tt-metal/issues/7159)

### CODE REVIEW

File: models/demos/t3000/mixtral8x7b/demo/demo.py [hunk 1]
```diff
@@ -220,7 +220,7 @@ def run_mixtral_demo(user_input, batch_size, device_mesh, instruct_mode):
 
         # Run ttnn mixtral model
         tt_out_11BH = tt_model(decode_input_11BH, start_pos, current_pos, attn_mask, rot_mats)
-
+        del decode_input_11BH, attn_mask
```
Comment (yieldthought) [inline models/demos/t3000/mixtral8x7b/demo/demo.py:223]:
```suggestion
        # Work around program cache issue https://github.com/tenstorrent/tt-metal/issues/7159
        del decode_input_11BH, attn_mask
```

File: models/demos/t3000/mixtral8x7b/tests/test_mixtral_attention.py [hunk 2]
```diff
@@ -73,6 +73,7 @@ def test_mixtral_attention_inference(t3k_device_mesh, use_program_cache, reset_s
             attn_mask,
             rot_mat,
         )
+        del attention_input, attn_mask
```
Comment (yieldthought) [inline models/demos/t3000/mixtral8x7b/tests/test_mixtral_attention.py:76]:
```suggestion
        # Work around program cache issue https://github.com/tenstorrent/tt-metal/issues/7159
        del attention_input, attn_mask
```

File: models/demos/t3000/mixtral8x7b/tests/test_mixtral_decoder.py [hunk 3]
```diff
@@ -77,6 +77,7 @@ def test_mixtral_decoder_inference(t3k_device_mesh, use_program_cache, reset_see
         )
         # Run TT model
         tt_out_b1sh = tt_model(decode_input_b1sh, start_pos, current_pos, attn_mask, rot_mat)
+        del decode_input_b1sh, attn_mask
```
Comment (yieldthought) [inline models/demos/t3000/mixtral8x7b/tests/test_mixtral_decoder.py:80]:
```suggestion
        # Work around program cache issue https://github.com/tenstorrent/tt-metal/issues/7159
        del decode_input_b1sh, attn_mask
```

File: models/demos/t3000/mixtral8x7b/tests/test_mixtral_model.py [hunk 4]
```diff
@@ -125,7 +125,7 @@ def test_mixtral_model_inference(
 
         # Run TT model
         tt_out = tt_model(decode_input, start_pos, current_pos, attn_mask, rot_mat)
-
+        del decode_input, attn_mask
```
Comment (yieldthought) [inline models/demos/t3000/mixtral8x7b/tests/test_mixtral_model.py:128]:
```suggestion
        # Work around program cache issue https://github.com/tenstorrent/tt-metal/issues/7159
        del decode_input, attn_mask
```

File: models/demos/t3000/mixtral8x7b/tests/test_mixtral_perf.py [hunk 5]
```diff
@@ -166,7 +166,7 @@ def run_inference(tt_model, embd, encoded_prompts, generation_start_pos, generat
         # Run TT model
         profiler.start(f"model_run_for_inference_{i}")
         tt_out = tt_model(decode_input, start_pos, current_pos, attn_mask, rot_mat)
-
+        del decode_input, attn_mask
```
Comment (yieldthought) [inline models/demos/t3000/mixtral8x7b/tests/test_mixtral_perf.py:169]:
```suggestion
        # Work around program cache issue https://github.com/tenstorrent/tt-metal/issues/7159
        del decode_input, attn_mask
```

**Review by yieldthought — CHANGES_REQUESTED** [review]
Let's start documenting our workarounds clearly so:
1. Other people reading the code know why it is there. These people might be us in the future ^^
2. We know and can easily search for which lines of code to remove when an issue is resolved

**Review by yieldthought — APPROVED** [review]

**Review by mtairum — APPROVED** [review]

**Pull request merged**